File: app/src/main/java/com/lib/custom/toast/Sample.java
package com.lib.custom.toast;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.ib.custom.toast.CustomToast;

/*
 * Sample App on How To Use Library
 */

public class Sample extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.layout);
        initialiseViews();
    }

    /**
     * initialise views
     */
    private void initialiseViews() {

        Button btnDefault = findViewById(R.id.btnDefault);
        btnDefault.setOnClickListener(this);

        Button btnSuccess = findViewById(R.id.btnSuccess);
        btnSuccess.setOnClickListener(this);

        Button btnError = findViewById(R.id.btnError);
        btnError.setOnClickListener(this);

        Button btnWarning = findViewById(R.id.btnWarning);
        btnWarning.setOnClickListener(this);

        Button btnInfo = findViewById(R.id.btnInfo);
        btnInfo.setOnClickListener(this);

    }

    /**
     *
     * @param view the view which has been clicked
     */
    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btnDefault:
                CustomToast.makeText(this, Toast.LENGTH_SHORT,CustomToast.DEFAULT, "Toast is working",false).show();
                break;

            case R.id.btnSuccess:
                CustomToast.makeText(this, Toast.LENGTH_SHORT,CustomToast.SUCCESS,"Toast is working",false).show();
                break;

            case R.id.btnError:
                CustomToast.makeText(this, Toast.LENGTH_SHORT,CustomToast.ERROR,"Username is not valid",false).show();
                break;

            case R.id.btnWarning:
                CustomToast.makeText(this, Toast.LENGTH_SHORT,CustomToast.WARNING,"Out Of Memory",false).show();
                break;

            case R.id.btnInfo:
                CustomToast.makeText(this, Toast.LENGTH_SHORT,CustomToast.INFO,"This is a customised Toast",false).show();
                break;
        }
    }
}

File: app/src/main/java/com/lib/custom/toast/Sample.kt
package com.lib.custom.toast

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.Toast
import com.ib.custom.toast.CustomToastView

/*
 * Check commented SampleToastActivity for java code
 */
class Sample :AppCompatActivity(), View.OnClickListener {

    private lateinit var btDefault:Button

    private lateinit var btnError:Button

    private lateinit var btnWarning:Button

    private lateinit var btnInfo:Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sample_toast)
        initialiseViews()
    }

    private fun initialiseViews(){
        btDefault = findViewById(R.id.btnDefault)
        btDefault.setOnClickListener(this)

        btnError = findViewById(R.id.btnError)
        btnError.setOnClickListener(this)

        btnWarning = findViewById(R.id.btnWarning)
        btnWarning.setOnClickListener(this)

        btnInfo = findViewById(R.id.btnInfo)
        btnInfo.setOnClickListener(this)
    }


    override fun onClick(p0: View?) {
        when(p0?.id)
        {
            R.id.btnDefault -> CustomToastView.makeText(this@Sample,Toast.LENGTH_SHORT,"Toast is working",R.layout.toast).show()

            R.id.btnSuccess -> CustomToastView.makeText(this@Sample,Toast.LENGTH_SHORT,CustomToastView.SUCCESS,"Toast is working",false).show()

            R.id.btnError   -> CustomToastView.makeText(this@Sample,Toast.LENGTH_SHORT,CustomToastView.ERROR,"Username is not valid",false).show()

            R.id.btnWarning -> CustomToastView.makeText(this@Sample,Toast.LENGTH_SHORT,CustomToastView.WARNING,"Out of Memory",false).show()

            R.id.btnInfo    -> CustomToastView.makeText(this@Sample, Toast.LENGTH_SHORT,CustomToastView.INFO,"password should be atleast 6 characters long",false).show()
        }
    }

}